### PR TITLE
Datasource: user

### PR DESCRIPTION
**Pull request by vmanilo**
Resolves https://github.com/Twingate/terraform-provider-twingate/issues/61

## Changes
- added datasource for twingate user

### CODE REVIEW

File: twingate/datasource_user.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+			},
+			"email": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The email of the User",
+			},
+			"is_admin": {
+				Type:        schema.TypeBool,
+				Computed:    true,
+				Description: "Indicates if the User admin or not",
```
Comment (vmanilo) [inline twingate/datasource_user.go:72]: @alexmensch please check descriptions

**Comment by vmanilo** [comment]
need to add user role

File: docs/data-sources/user.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  id = "<your user's id>"
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `id` (String) The ID of the User
```
Comment (alexmensch) [inline docs/data-sources/user.md:26]: Update to:

`...User. The ID for the User must be obtained from the Admin API.`

File: docs/data-sources/user.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "twingate_user Data Source - terraform-provider-twingate"
+subcategory: ""
+description: |-
+  Twingate users. For more information, see Twingate's documentation https://docs.twingate.com/docs/users.
+---
+
+# twingate_user (Data Source)
+
+Twingate users. For more information, see Twingate's [documentation](https://docs.twingate.com/docs/users).
```
Comment (alexmensch) [inline docs/data-sources/user.md:11]: Let's expand the description to the following:

`Users in Twingate can be given access to Twingate Resources and may either be added manually or automatically synchronized with a 3rd party identity provider. For more information, see see Twingate's [documentation](https://docs.twingate.com/docs/users).`

**Review by alexmensch — CHANGES_REQUESTED** [review]

File: docs/data-sources/user.md (hunk 2 above)
Comment (vmanilo) [reply docs/data-sources/user.md]: updated

File: docs/data-sources/user.md (hunk 3 above)
Comment (vmanilo) [reply docs/data-sources/user.md]: updated

**Review by ekampf — APPROVED** [review]

File: docs/data-sources/user.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `id` (String) The ID of the User
+
+### Read-Only
+
+- `email` (String) The email of the User
```
[the remark this answers is not in the document]
Comment (alexmensch) [reply docs/data-sources/user.md:30]: Hey @vmanilo I think this still needs to be addressed?

File: docs/data-sources/user.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+- `first_name` (String) The first name of the User
+- `is_admin` (Boolean) Indicates if the User admin or not
```
[the remark this answers is not in the document]
Comment (alexmensch) [reply docs/data-sources/user.md:32]: And this one, too, @vmanilo

**Review by alexmensch — CHANGES_REQUESTED** [review]
Changes still required to docs, see comment thread

File: docs/data-sources/user.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (alexmensch) [reply docs/data-sources/user.md:32]: Also for the `is_admin` attribute, it should read: `Indicates whether the User is an admin`

File: docs/data-sources/user.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (vmanilo) [reply docs/data-sources/user.md]: updated

File: docs/data-sources/user.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (vmanilo) [reply docs/data-sources/user.md]: updayed

**Review by alexmensch — APPROVED** [review]

**Pull request merged**